### PR TITLE
Fix Content-Type warning

### PR DESCRIPTION
**Pull request by chrisroos**
This change removes the following warning we were seeing when running the tests in this project, and when using the `AssetManager#create_asset`, `#create_whitehall_asset` and `#update_asset` methods in other project:

```
warning: Overriding "Content-Type" header "application/json" with "multipart/form-data; boundary=----RubyFormBoundaryX7Na6WDQqG3kLfD7" due to payload
```

Addresses issue #757.

### CODE REVIEW

File: lib/gds_api/json_client.rb [hunk 1]
```diff
@@ -246,7 +246,11 @@ def do_request(method, url, params = nil, additional_headers = {})
       when :get, :delete
         default_headers = self.class.default_request_headers
       else
-        default_headers = self.class.default_request_with_json_body_headers
+        if params.respond_to?(:has_key?) && params[:multipart] == true
```
Comment (floehopper) [inline lib/gds_api/json_client.rb]: Minor: It seems slightly odd to check for the existence of the `has_key?` method. Presumably you can't use `params.respond_to?(:[])` here because a `String` also responds to that method...? If so, I think it might make more sense to use `params.is_a?(Hash)` in this case.

File: lib/gds_api/json_client.rb [hunk 2]
```diff
@@ -242,20 +239,9 @@ def do_request(method, url, params = nil, additional_headers = {})
         url: url,
       }
 
-      case method
-      when :get, :delete
```
Comment (floehopper) [inline lib/gds_api/json_client.rb]: Minor: I found this commit a bit hard to follow. I think it might help to explain why you no longer need this condition and how come all the tests still pass.

File: lib/gds_api/json_client.rb (hunk 1 above)
Comment (floehopper) [reply lib/gds_api/json_client.rb]: It looks like this code is deleted in the next commit, so you can probably ignore my previous comment.

**Comment by chrisroos** [comment]
Thanks for reviewing, @floehopper. I'm going to address your feedback now.

File: lib/gds_api/json_client.rb (hunk 2 above)
Comment (chrisroos) [reply lib/gds_api/json_client.rb]: I've updated the commit message in an attempt to better explain the change.

**Comment by chrisroos** [comment]
> You might want to incorporate a version bump in this PR otherwise you'll need to open another one to get a new version of the gem released.

I'm going to do this in another branch/PR as per the instructions in https://github.com/alphagov/styleguides/blob/master/rubygems.md#versioning.

File: lib/gds_api/json_client.rb (hunk 2 above)
Comment (floehopper) [reply lib/gds_api/json_client.rb]: Looks great! 👍
